src/main/java/com/minecolonies/coremod/colony/buildings/BuildingDeliveryman.java

```diff
@@ -33,26 +36,39 @@ public BuildingDeliveryman(final Colony c, final BlockPos l)
         super(c, l);
     }
 
-    @NotNull
-    @Override
-    public String getSchematicName()
+    /**
+     * Get the building the deliveryman should deliver to.
+     *
+     * @return the building.
+     */
+    public AbstractBuilding getBuildingToDeliver()
     {
-        return DELIVERYMAN;
+        return this.buildingToDeliver;
     }
 
-    @Override
-    public int getMaxBuildingLevel()
+    /**
+     * Set the building the deliveryman should deliver to.
+     *
+     * @param building building to deliver to.
+     */
+    public void setBuildingToDeliver(AbstractBuilding building)
     {
-        return 4;
+        this.buildingToDeliver = building;
     }
 
     @NotNull
     @Override
-    public String getJobName()
+    public String getSchematicName()
     {
         return DELIVERYMAN;
     }
 
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return 5;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)